cmd/kubectl-azad-proxy/actions/discover.go

```diff
@@ -216,6 +222,82 @@ func (client *DiscoverClient) Run(ctx context.Context) ([]discover, error) {
 	return discoverData, nil
 }
 
+func (client *DiscoverClient) trySubscriptionsDiscovery(ctx context.Context) ([]discover, error) {
+	log := logr.FromContextOrDiscard(ctx)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, customerrors.New(customerrors.ErrorTypeAuthorization, err)
+	}
+
+	subscriptionClient, err := armsubscriptions.NewClient(cred, nil)
+	if err != nil {
+		return nil, customerrors.New(customerrors.ErrorTypeAuthorization, err)
+	}
+
+	subscriptionsIds := []string{}
+	pager := subscriptionClient.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, customerrors.New(customerrors.ErrorTypeAuthorization, err)
+		}
+		for _, v := range nextResult.Value {
+			if v.SubscriptionID == nil {
+				continue
+			}
+			subscriptionsIds = append(subscriptionsIds, *v.SubscriptionID)
+		}
+	}
+
+	for _, subscriptionId := range subscriptionsIds {
+		log.V(1).Info("Trying to find clusters on subscription", "subscription_id", subscriptionId)
+		clusters, err := client.trySubscriptionDiscovery(ctx, cred, subscriptionId)
+		if err == nil {
+			return clusters, nil
+		}
+	}
+
+	return nil, customerrors.New(customerrors.ErrorTypeAuthentication, fmt.Errorf("unable to find any clusters on any subscriptions"))
+}
+
+func (client *DiscoverClient) trySubscriptionDiscovery(ctx context.Context, cred *azidentity.DefaultAzureCredential, subscriptionId string) ([]discover, error) {
+	tagClient, err := armresources.NewClient(subscriptionId, cred, nil)
+	if err != nil {
+		return nil, err
+	}
+	res, err := tagClient.GetByID(ctx, fmt.Sprintf("/subscriptions/%s", subscriptionId), "2021-04-01", &armresources.ClientGetByIDOptions{})
```

Comment:
Should we make 2021-04-01 in to a constant?

Reply:
Fixed!